### PR TITLE
Swap to new model

### PR DESCRIPTION
**Pull request by jvshields**
### :pencil: Description

**Type:**  :rocket: `feature` | :biohazard: `breaking change` | :roller_coaster: `infrastructure`

This PR attaches swaps the old model created by read_marcs_to_fv() out and hooks up the new model reader to the run_stardis() function. This effectively implements much of the restructure work with the new infrastructure, and so it also includes changing the code in many places to read information from the new model structure. 

Future PRs need to change the Configuration schema to allow for general model reading, as the current code only allows for unzipped MARCS models, so the configuration should allow to specify whether the input model is gzipped or not. 

Additionally, I have left many comments around the code pointing out parts that need cleaning or changing, because they are misleading or redundant. These are not-functional changes, and should be changed in a future cleaning PR. 

I am leaving the read_marcs_to_fv() function for now, because it is currently integrated into the benchmarking pipeline, but a future PR should also swap that over to the new structure as well.

### :vertical_traffic_light: Testing

How did you test these changes?
I have tested the output of the quickstart notebook with the same model as before. I get a different answer for the spectrum, shown here

Comparison Spectrum

There is a very small difference here that is not entirely unexpected. The previous model reader made physical changes to the system by naively interpolating the temperatures and densities of the system and then defining cells with those values, and solving an opacity for those cells. This new method solves the plasma using the values provided by the model, and then linearly interpolates the opacities measured at each of the points for light to travel from one depth point to the next. I generally believe that the new model is working correctly, but as far as I can tell it's impossible to directly reproduce the previous model because that model solves a completely different plasma and then solves the radiative transfer equation differently. 

The answers are close enough that I think the new model is working correctly. 

- [x] Testing pipeline
- [x] Other method (describe)

### :ballot_box_with_check: Checklist

- [ ] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

File: stardis/io/model/marcs.py [hunk 1]
```diff
@@ -75,32 +79,45 @@ def convert_marcs_raw_abundances_to_mass_fractions(
...
         for atom_num, col in enumerate(marcs_chemical_mass_fractions.columns):
             if atom_num < len(atom_data.atom_data):
-                marcs_chemical_mass_fractions[f"mass_fraction_{atom_num+1}"] = (
+                marcs_chemical_mass_fractions[(atom_num + 1)] = (
                     10 ** marcs_chemical_mass_fractions[col]
                 ) * atom_data.atom_data.mass.iloc[atom_num]
             else:
                 for j in range(atom_num, marcs_chemical_mass_fractions.shape[1]):
-                    marcs_chemical_mass_fractions[f"mass_fraction_{j+1}"] = np.nan
+                    marcs_chemical_mass_fractions[(j + 1)] = np.nan
```
Comment (sonachitchyan) [inline stardis/io/model/marcs.py:87]: why parentheses?

File: stardis/io/model/marcs.py [hunk 2]
```diff
@@ -75,32 +79,45 @@ def convert_marcs_raw_abundances_to_mass_fractions(
...
+        # Truncate to final atomic number, if smaller than the number of chemicals in the model
         marcs_chemical_mass_fractions = marcs_chemical_mass_fractions.iloc[
             :,
             : np.min([final_atomic_number, num_of_chemicals_in_model]),
         ]
 
-        marcs_chemical_mass_fractions = marcs_chemical_mass_fractions[::-1]
+        # Convert to atom data format expected by tardis plasma
+        marcs_atom_data = pd.DataFrame(
+            columns=marcs_chemical_mass_fractions.index
```
Comment (sonachitchyan) [inline stardis/io/model/marcs.py:110]: is there a way of doing this when defining ```marcs_chemical_mass_fractions```?

**Review by sonachitchyan — DISMISSED** [review]
Looks good! The only thing is maybe we should move the comments to an issue and make a separate PR for solving them? This way it feels disconnected from the purpose of this PR

File: stardis/io/model/marcs.py (hunk 2 above)
Comment (jvshields) [reply stardis/io/model/marcs.py:110]: It's a mismatch the numbered points in the model and what tardis expects. Not sure there's a cleaner transformation.

**Review by sonachitchyan — DISMISSED** [review]

**Review by andrewfullard — APPROVED** [review]

**Pull request merged**